### ISSUE TITLE
Clarification on YOLO v12 Output Tensor Format and Post-Processing

### ISSUE THREAD

**Issue by joaoscheuermann** [ultralytics/ultralytics#22697]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hi,

I am using the YOLO v12 model on a ONNX Runtime in Flutter for an Object Detection task and need some clarification regarding the raw output tensor to implement the correct post-processing.

My model output tensor shape is [1, 6, 8400] (which I transpose to [1, 8400, 6]). I have three specific questions about interpreting these values:

1 - **Coordinate Format:** Does the model output coordinates as [x_min, y_min, width, height] or [center_x, center_y, width, height]?

2 - **Class Scores:** Do the class confidence values (indices 4 and 5 in my case) represent raw logits that require a Sigmoid function, or are they already normalized probabilities?

3 - **Non-Maximum Suppression (NMS):** Is NMS applied within the exported model graph, or is it expected to be handled externally in post-processing?

Thank you for the help!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @joaoscheuermann, thank you for opening this issue and for using Ultralytics YOLO with ONNX Runtime in Flutter 🚀  
This is an automated assistant response to help you get started quickly — an Ultralytics engineer will also review your question and follow up soon.

Since your topic is about inference and post-processing (coordinate formats, class scores, and NMS behavior for your YOLO v12 ONNX model), please include as many concrete details as possible to help the team reproduce and reason about your setup, for example:

- The exact ONNX export command or steps you used (including any export arguments)
- The model variant you started from (e.g., YOLO11x, YOLO12 equivalent, etc.)
- A short code snippet showing:
  - How you run inference in ONNX Runtime
  - How you transpose and interpret the `[1, 6, 8400]` output
- A small example of raw model outputs (for 1–2 predictions) and how you are currently decoding them

This additional context will make it easier for the engineers to clarify the expected tensor format and post-processing details for your specific case 💡

In the meantime, you may find it helpful to review the general Ultralytics docs for reference on usage patterns and modes:

- [Docs](https://docs.ultralytics.com/)
- [Python examples](https://docs.ultralytics.com/usage/python/)
- [CLI examples](https://docs.ultralytics.com/usage/cli/)

If this evolves into a 🐛 Bug Report (for example, if you confirm a mismatch between documented and actual outputs), please also share a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) so the team can debug more efficiently.

For custom training or advanced inference ❓ questions, detailed information such as sample inputs/outputs, logs, and screenshots of incorrect detections are extremely helpful. Also, if you are training or fine-tuning models yourself, please verify you are following the [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

You’re also very welcome to connect with the Ultralytics community for extra support and discussion:
- Real-time chat on [Discord](https://discord.com/invite/ultralytics) 🎧  
- Longer-form questions on [Discourse](https://community.ultralytics.com/)  
- Community threads on the [Subreddit](https://www.reddit.com/r/Ultralytics/)  

## Upgrade

Before diving deeper, please make sure you are testing with the latest `ultralytics` package and requirements, in a supported environment, to rule out issues already fixed in recent versions:

```bash
pip install -U ultralytics
```

Refer to the [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) and use a [Python>=3.8](https://www.python.org/) environment with [PyTorch>=1.8](https://pytorch.org/get-started/locally/).

## Environments

YOLO can be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See the [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See the [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See the [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

These are useful for cross-checking behavior against your Flutter/ONNX Runtime pipeline to see if the outputs match expectations.

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
1. Depends on whether you exported with `nms`. See: https://www.reddit.com/r/Ultralytics/comments/1jwl6se/ultralytics_postprocessing_guide/
2. Sigmoid is already applied.
3. Depends on whether you exported with `nms`

**Comment by joaoscheuermann** [comment]
Thanks, I took a look on it and exported with NMS, now it's working properly.

**Comment by glenn-jocher** [comment]
Nice, glad the ONNX export with `nms=True` solved it; for future reference the NMS-wrapped export returns per-detection `[x1, y1, x2, y2, score, class]` as in the [`NMSModel` export wrapper docs](https://docs.ultralytics.com/reference/engine/exporter/#nmsmodel) and the Detect head [`postprocess` reference](https://docs.ultralytics.com/reference/nn/modules/head/#postprocess), and for new deployments I’d suggest considering a YOLO11 model since [YOLO11](https://docs.ultralytics.com/models/yolo11/) is the recommended, more stable option compared to YOLO12.
